File: Features/Ships/Base/Models/Ship.cs
using System.Collections.Generic;

namespace CorfuCruises {

    public class Ship {

        public int Id { get; set; }
        public int ShipOwnerId { get; set; }
        public string Description { get; set; }
        public string IMO { get; set; }
        public string Flag { get; set; }
        public string RegistryNo { get; set; }
        public string Manager { get; set; }
        public string ManagerInGreece { get; set; }
        public string Agent { get; set; }
        public bool IsActive { get; set; }
        public string UserId { get; set; }

        public ShipOwner ShipOwner { get; set; }

        public IEnumerable<Registrar> Registrars { get; set; }

    }

}

File: FrontEnd/src/app/features/ships/base/classes/models/ship.ts
import { Crew } from 'src/app/features/shipCrews/classes/models/crew'
import { Registrar } from '../../../registrars/classes/registrar'
import { ShipOwner } from '../../../owners/classes/base/ship-owner'

export class Ship {

    id: number
    description: string
    imo: string
    flag: string
    registryNo: string
    manager: string
    managerInGreece: string
    agent: string
    isActive: boolean

    shipOwner: ShipOwner
    registrars: Registrar[] = []
    crew: Crew[] = []

}
